src/config/yaml.rs

```diff
@@ -108,19 +109,36 @@ impl YamlConfig {
         serde_yaml::from_reader(reader).map_err(|e| e.into())
     }
 
-    pub fn get(env: &EnvConfig) -> Result<Self> {
+    pub fn get(env: &EnvConfig) -> Result<YamlConfig> {
         if let Some(yaml) = env.config_yaml.as_ref() {
-            return Self::from_str(yaml);
+            // We're getting the configuration from the environment variable `NAVI_CONFIG_YAML`
+            let mut cfg = Self::from_str(yaml)?;
+            cfg.source = format!("NAVI_CONFIG_YAML={}", yaml);
+
+            return Ok(cfg);
         }
         if let Some(path_str) = env.config_path.as_ref() {
+            // We're getting the configuration from a file given in the environment variable 'NAVI_CONFIG'
+
             let p = PathBuf::from(path_str);
-            return YamlConfig::from_path(&p);
+            let mut cfg = YamlConfig::from_path(&p)?;
+            cfg.source = format!("NAVI_CONFIG_YAML={:?}", p).to_string();
```

Comment:
```suggestion
            cfg.source = format!("{}={}", env_var::CONFIG, p.to_string());
```
`NAVI_CONFIG_YAML` is typo of `NAVI_CONFIG` and it is better to use `env_var::CONFIG` that is used in `env.config_path`.
It is better to avoid using `{:?}` for anything other than debugging.
Unnecessary `to_string()`.